src/telliot_feeds/reporters/reporter_autopay_utils.py

```diff
@@ -311,21 +311,26 @@ async def get_feed_tip(query: bytes, autopay: TellorFlexAutopayContract) -> Opti
         msg = "can't suggest feed, autopay contract not connected"
         error_status(note=msg, log=logger.critical)
         return None
-    try:
+
+    if query in CATALOG_QUERY_IDS:
         single_query = {query: CATALOG_QUERY_IDS[query]}
-    except KeyError:
-        qtype_name, _ = decode_single("(string,bytes)", query)
-        if qtype_name not in Registry.registry:
-            logger.warning(f"Unsupported query type: {qtype_name}")
+    else:
+        try:
+            qtype_name, _ = decode_single("(string,bytes)", query)
+            if qtype_name not in Registry.registry:
+                logger.warning(f"Unsupported query type: {qtype_name}")
+                return None
+            else:
+                query = Web3.keccak(query)
+                CATALOG_QUERY_IDS[query] = query.hex() + qtype_name.lower()
+                single_query = {query: CATALOG_QUERY_IDS[query]}
+        except OverflowError:
```

Comment:
Your specific exceptions should wrap only the line that would generate the error, so put after line 319 I think

Reply:
will do